crates/sui-json-rpc-types/src/sui_event.rs

```diff
@@ -122,6 +126,65 @@ impl SuiEvent {
     }
 }
 
+impl Display for SuiEvent {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let parsed_json = &mut self.parsed_json.clone();
+        bytes_array_to_base64(parsed_json);
+        let mut table = json_to_table(parsed_json);
+        let style = TableStyle::modern();
+        table.collapse().with(style);
+        let bcs_base58 = Base58::encode(&self.bcs);
+        write!(f,
+            " ┌──\n │ EventID: {}:{}\n │ PackageID: {}\n │ Transaction Module: {}\n │ Sender: {}\n │ EventType: {}\n │ BCS (Base58):{}\n",
+            self.id.tx_digest, self.id.event_seq, self.package_id, self.transaction_module, self.sender, self.type_, bcs_base58)?;
+        if let Some(ts) = self.timestamp_ms {
+            writeln!(f, " │ Timestamp: {}\n └──", ts)?;
+        }
+        writeln!(f, " │ ParsedJSON:")?;
+        let table_string = table.to_string();
+        let table_rows = table_string.split_inclusive('\n');
+        for r in table_rows {
+            write!(f, " │   {r}")?;
+        }
+
+        write!(f, "\n └──")
+    }
+}
+
+// Transform in-place a JSON array of bytes into a Base64 string
+fn bytes_array_to_base64(v: &mut Value) {
+    match v {
+        Value::Null | Value::Bool(_) | Value::Number(_) | Value::String(_) => (),
+        Value::Array(ref mut vals) => {
+            if vals.iter().all(check_is_number) {
+                let new_vals = vals
+                    .iter()
+                    .map(|num| num.as_u64().unwrap() as u8)
+                    .collect::<Vec<_>>();
+                let new_val = serde_json::json!(Base64::from_bytes(&new_vals).encoded());
+                *v = new_val;
+            } else {
+                for val in vals {
+                    bytes_array_to_base64(val)
+                }
+            }
+        }
+        Value::Object(map) => {
+            for val in map.values_mut() {
+                bytes_array_to_base64(val)
+            }
+        }
+    }
+}
+
+fn check_is_number(v: &Value) -> bool {
+    if let Some(num) = v.as_u64() {
+        num <= 255u64
+    } else {
+        false
+    }
+}
```

Comment:
Couple of nits for these functions:

- You may not need the `ref mut` on `vals`?  My intuition on this is pretty poor, but try removing it and see what happens, because you don't need it for the conversion to a byte array, the recursive case for `Object` seems to not need it either, and you are already matching on a mutable reference, so I think `vals` should already be a `&mut`.
- You can avoid the `unwrap` on `as_u64` (and calling `as_u64` twice), by taking advantage of the fact that if you have an iterator of `Option<T>` you can collection it into an `Option<Vec<T>>`.
- `check_is_number` is quite a generic name -- you also care that it's a byte.
- `use serde_json::json`.

```suggestion
fn bytes_array_to_base64(v: &mut Value) {
    match v {
        Value::Null | Value::Bool(_) | Value::Number(_) | Value::String(_) => (),
        Value::Array(vals) => {
            if let Some(vals) = vals.iter().map(try_into_byte).collect::<Option<Vec<_>>>() {
                *v = json!(Base64::from_bytes(&vals).encoded())
            } else {
                for val in vals {
                    bytes_array_to_base64(val)
                }
            }
        }
        Value::Object(map) => {
            for val in map.values_mut() {
                bytes_array_to_base64(val)
            }
        }
    }
}

fn try_into_byte(v: &Value) -> Option<u8> {
    let num = v.as_u64()?;
    (num <= 255).then_some(num as u8)
}
```

Reply:
Indeed, that worked nicely. Thanks for making the code a lot more elegant!